finat/sympy2gem.py

```diff
@@ -65,3 +66,72 @@ def sympy2gem_symbol(node, self):
 @sympy2gem.register(symengine.Rational)
 def sympy2gem_rational(node, self):
     return gem.Division(*(map(self, node.as_numer_denom())))
+
+
+@sympy2gem.register(sympy.Abs)
+@sympy2gem.register(symengine.Abs)
+def sympy2gem_abs(node, self):
+    return gem.MathFunction("abs", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Not)
+@sympy2gem.register(symengine.Not)
+def sympy2gem_not(node, self):
+    return gem.LogicalNot(*map(self, node.args))
+
+
+@sympy2gem.register(sympy.Or)
+@sympy2gem.register(symengine.Or)
+def sympy2gem_or(node, self):
+    return reduce(gem.LogicalOr, map(self, node.args))
+
+
+@sympy2gem.register(sympy.And)
+@sympy2gem.register(symengine.And)
+def sympy2gem_and(node, self):
+    return reduce(gem.LogicalAnd, map(self, node.args))
+
+
+@sympy2gem.register(sympy.Eq)
+@sympy2gem.register(symengine.Eq)
+def sympy2gem_eq(node, self):
+    return gem.Comparison("==", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Gt)
+def sympy2gem_gt(node, self):
+    return gem.Comparison(">", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Ge)
+def sympy2gem_ge(node, self):
+    return gem.Comparison(">=", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Lt)
+@sympy2gem.register(symengine.Lt)
+def sympy2gem_lt(node, self):
+    return gem.Comparison("<", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Le)
+@sympy2gem.register(symengine.Le)
+def sympy2gem_le(node, self):
+    return gem.Comparison("<=", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Piecewise)
+@sympy2gem.register(symengine.Piecewise)
+def sympy2gem_conditional(node, self):
+    expr = None
+    pieces = []
+    for v, c in node.args:
+        if isinstance(c, (bool, numpy.bool)) and c:
```

Comment:
@dham we also need to consider `sympy.logic.boolalg.BooleanTrue`, should we revert to `c == True`?